src/core/qgshelp.h

```diff
@@ -0,0 +1,73 @@
+/***************************************************************************
+  qgshelp.h
+  --------------------------------------
+  Date                 : December 2016
+  Copyright            : (C) 2016 by Alexander Bruy
+  Email                : alexander dot bruy at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSHELP_H
+#define QGSHELP_H
+
+#include <QObject>
+
+/** \ingroup core
+ * @brief The class provides help URI for the given key.
+ *
+ * Help can be stored online, on the local directory or on the intranet
+ * server. Location of the QGIS help can be configured in QGIS options.
+ * Multiple locations are supported, they will be used in order of
+ * preference, from top to bottom.
+ *
+ * URI construction takes in account following information:
+ *  - QGIS version
+ *  - language of the QGIS UI
+ *
+ * If no help found, default error page with information how to setup
+ * help system will be shown.
+ *
+ * @note added in QGIS 3.0
+ */
+class CORE_EXPORT QgsHelp : public QObject
+{
+    Q_OBJECT
+  public:
+    /** Opens help topic for the given help key using default system
+     * web browser. If help topic not found, builtin error page shown.
+     * @param key key which identified help topic
+     * @note added in QGIS 3.0
+     */
+    static void openHelp( const QString& key );
```

Comment:
Following from the earlier comment about not using a singleton here, these methods would become const members rather than static methods.